packages/patternfly-3/patternfly-react/src/components/ListView/ListViewItem.js

```diff
@@ -15,7 +15,12 @@ import ListViewRow from './ListViewRow';
  * renders ListViewGroupItemHeader and ListViewGroupItemContainer
  */
 class ListViewItem extends React.Component {
-  state = { expanded: false };
+  constructor(props) {
+    super(props);
+
+    this.state = { expanded: props.expanded };
```

Comment:
this is a good use case for `getDerivedStateFromProps` I believe...

Reply:
@priley86 hmm, will read the docs, thanks for the pointer! https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops

I am a bit new to react.js :)

Reply:
i like this methodology for simple ui state (i.e list item expansion, tree item expansion, etc), simply b/c, we can manage the default behavior in local state for the user, but still allow them to override via prop at any point explicitly if desired...(so state mirrors prop, can be initialized in the constructor, and can be updated explicitly later)... kind of like the example w/ `activeStepIndex` here:
https://github.com/patternfly/patternfly-react/blob/a272045e4861c450cc7158d2a717da20a7e7c39a/packages/patternfly-3/patternfly-react/src/components/Wizard/Patterns/StatefulWizardPattern.js#L14

Reply:
Provided another example of this here:
https://github.com/patternfly/patternfly-react/pull/613#issuecomment-427935541

The `propOrState` helper is not necessarily the same thing, but could be used to do something similar. For expansion state, you may prefer to just mirror the state and update only when the prop changes...

Reply:
thanks, will take a look, this got a bit slipped lower on my todo list, but I will get to it :)